src/ngraph/op/embeddingbag_offsets_sum.cpp

```diff
@@ -0,0 +1,167 @@
+        get_input_element_type(OFFSETS),
+        ") must match indices element type (",
+        get_input_element_type(INDICES),
+        ")");
+
+    NODE_VALIDATION_CHECK(this,
+                          get_input_partial_shape(INDICES).is_dynamic() ||
+                              get_input_partial_shape(INDICES).to_shape().size() == 1,
+                          "INDICES must be 1D");
+
+    NODE_VALIDATION_CHECK(this,
+                          get_input_partial_shape(OFFSETS).is_dynamic() ||
+                              get_input_partial_shape(OFFSETS).to_shape().size() == 1,
+                          "OFFSETS must be 1D");
+
+    if (get_input_size() >= 4)
+    {
+        NODE_VALIDATION_CHECK(this,
+                              get_input_element_type(EMB_TABLE).compatible(
+                                  get_input_element_type(PER_SAMPLE_WEIGHTS)),
+                              "Per sample weight element type (",
+                              get_input_element_type(PER_SAMPLE_WEIGHTS),
+                              ") must match embedding table element type (",
+                              get_input_element_type(EMB_TABLE),
+                              ")");
+
+        NODE_VALIDATION_CHECK(this,
+                              get_input_partial_shape(PER_SAMPLE_WEIGHTS).is_dynamic() ||
+                                  get_input_partial_shape(PER_SAMPLE_WEIGHTS).to_shape().size() ==
+                                      1,
+                              "PER_SAMPLE_WEIGHTS must be 1D");
+
+        NODE_VALIDATION_CHECK(this,
+                              get_input_partial_shape(INDICES).compatible(
+                                  get_input_partial_shape(PER_SAMPLE_WEIGHTS)),
+                              "INDICES and PER_SAMPLE_WEIGHTS shape must be same");
+    }
+
+    if (get_input_size() == 5)
+    {
+        NODE_VALIDATION_CHECK(this,
+                              get_input_element_type(DEFAULT_INDEX) == element::i64 ||
+                                  get_input_element_type(DEFAULT_INDEX) == element::i32,
+                              "DEFAULT_INDEX type must be i32 or i64");
+
+        NODE_VALIDATION_CHECK(
+            this,
+            get_input_element_type(INDICES).compatible(get_input_element_type(DEFAULT_INDEX)),
+            "Default_index element type (",
+            get_input_element_type(DEFAULT_INDEX),
+            ") must match indices element type (",
+            get_input_element_type(INDICES),
+            ")");
+
+        NODE_VALIDATION_CHECK(this,
+                              get_input_partial_shape(DEFAULT_INDEX).compatible(PartialShape{}),
+                              "DEFAULT_INDEX must be a scalar");
+    }
+
+    element::Type result_et = get_input_element_type(EMB_TABLE);
+
+    const PartialShape& emb_table_shape = get_input_partial_shape(EMB_TABLE);
+    const PartialShape& offsets_shape = get_input_partial_shape(OFFSETS);
+
+    PartialShape result_shape;
```

Comment:
Why don't you add a `PartialShape(const Rank&)` constructor and then you (and others) can skip doing things like `result_dims`